api/wb/views.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import generics, status
+from rest_framework.exceptions import NotFound
+
+from osf.models import AbstractNode
+from addons.osfstorage.models import OsfStorageFileNode, OsfStorageFolder
+from api.base.utils import get_object_or_error
+from api.base.parsers import HMACSignedParser
+from api.wb.serializers import (
+    WaterbutlerMetadataSerializer
+)
+
+class FileMetadataMixin(generics.CreateAPIView):
```

Comment:
I'm not sure `CreateAPIView` is the right generic class. It's also non-intuitive that `WaterbutlerMetadataSerializer.create` doesn't actually create anything and contains logic for both copying and moving.

We can improve the design by putting the different actions in the different view classes.

Ideally it would look something like

```python
class MoveFileMetadataView(FileMetadataView):
    def perform_file_action(self, source, destination, name):
        return source.move_under(destination, name)

class CopyFileMetadataView(FileMetadataView):
    def perform_file_action(self, source, destination, name):
        return source.copy_under(destination, name)
```

The hypothetical `FileMetadataView` would do the work of parsing the payload returning the response, etc.